src/blocks.jl

```diff
@@ -92,3 +92,43 @@ This is broken for now. See: https://github.com/JuliaArrays/BlockArrays.jl/issue
     copyto!(a[i...], b)
     a
 end
+
+"""
+    blocksizes(A::AbstractArray)
+
+Return an iterator over the sizes of each block.
+See also size and blocksize.
+
+# Examples
+```jldoctest
+julia> A = BlockArray(ones(3,3),[2,1],[1,1,1])
+2×3-blocked 3×3 BlockMatrix{Float64}:
+ 1.0  │  1.0  │  1.0
+ 1.0  │  1.0  │  1.0
+ ─────┼───────┼─────
+ 1.0  │  1.0  │  1.0
+
+julia> collect(blocksizes(A))
+2×3 Matrix{Tuple{Int64, Int64}}:
+ (2, 1)  (2, 1)  (2, 1)
+ (1, 1)  (1, 1)  (1, 1)
+
+julia> blocksizes(A)[2, 2]
+(1, 1)
+```
+"""
+blocksizes(A::AbstractArray) = BlockSizes(A)
+
+struct BlockSizes{N,A<:AbstractArray{<:Any,N}}
+    array::A
+end
+Base.size(bs::BlockSizes) = blocksize(bs.array)
```

Comment:
We import `Base: size, length, axes, …` and the rest of the package doesn't include the "Base." in the overloads. I'd suggest keeping things consistent.

Reply:
Will do.